examples/hpc/dla-driver/src/tensor4.rs

```diff
@@ -324,8 +321,38 @@ impl<T: Clone> Tensor4<T> {
         if order == self.order {
             return self.to_buffer();
         }
+
+        // NOTE:(20240925 vaino-waltteri.granat@tuni.fi) TVM order fix
+        if self.order == Order4::HWCK && order == Order4::HWKC {
+            return self.tvm_layout_to_headsail();
+        }
+
         let mut data = self.clone();
         data.permute(order);
         data.to_buffer()
     }
+
+    /// Convert HWIO (HWCK) order to HWOI (HWKC) for headsail
+    pub fn tvm_layout_to_headsail(&self) -> Vec<T> {
+        let data = self.to_buffer();
+
+        let mut hwoi_flat =
+            Vec::with_capacity(self.height() * self.width() * self.channels() * self.kernels());
+        // Iterate and assign values from HWIO to HWOI format
+        for h_idx in 0..self.height() {
+            for w_idx in 0..self.width() {
+                for o_idx in 0..self.kernels() {
+                    for i_idx in 0..self.channels() {
+                        let hwio_index = h_idx * self.width() * self.channels() * self.kernels()
+                            + w_idx * self.channels() * self.kernels()
+                            + i_idx * self.kernels()
+                            + o_idx;
+                        hwoi_flat.push(data[hwio_index].clone());
+                    }
+                }
+            }
+        }
+
+        hwoi_flat
+    }
```

Comment:
Yes, I tried to implement this functionality with `permuted_axes()`, but I observed that it didn't work as I would have liked. It might be something with my implementation. For now, I will leave this as it is, and come back to it later if there is a need for other orders to work.